### PR TITLE
Make sure end date is reindexed when resolving/reactivating a dossier.

### LINKED ISSUE #5458: End date not reindexed correctly when reopening dossiers.
When reopening a dossiers the end-date index is not updated correctly in the catalog. Metadata are correct.

Reproduce:
- Create dossier 
- Close dossier
- Re-open dossier

Metadata:
screenshot 2019-03-06 at 13 19 24

Vs. catalog index:
screenshot 2019-03-06 at 13 19 19

This has surfaced in https://extranet.4teamwork.ch/support/gever-st-gallen/tracker/610

### PR DESCRIPTION
**Pull request by njohner**
⚠️ do not merge, missing upgrade ⚠️ 
resolves #5458

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog
 2019.2.0 (unreleased)
 ---------------------
 
+- Make sure end date is reindex when resolving/reactivating a dossier. [njohner]
```
Comment (deiferni) [inline docs/HISTORY.txt]: reindexed

File: opengever/core/upgrades/20190307102922_reindex_dossier_end_date/upgrade.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class ReindexDossierEndDate(UpgradeStep):
+    """Reindex dossier end date when there is a mismatch between index and metadata
+    """
+
+    def __call__(self):
+        query = {'portal_type': 'opengever.dossier.businesscasedossier'}
+        catalog = self.portal.portal_catalog
+        end = catalog._catalog.getIndex("end")
```
Comment (njohner) [inline opengever/core/upgrades/20190307102922_reindex_dossier_end_date/upgrade.py]: It's much faster to get data from a single index than for all indexes...

**Comment by njohner** [comment]
Ok I added and tested the upgrade step. This should be ready to 🚢

**Comment by njohner** [comment]
Ok this is ready. Deif I included your suggestion to catch all dossiers. Sorry about that, I had actually planed to ask about that...

File: opengever/core/upgrades/20190307102922_reindex_dossier_end_date/upgrade.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from ftw.upgrade import UpgradeStep
+
+
+class ReindexDossierEndDate(UpgradeStep):
+    """Reindex dossier end date when there is a mismatch between index and metadata
+    """
+
```
Comment (deiferni) [inline opengever/core/upgrades/20190307102922_reindex_dossier_end_date/upgrade.py]: oh no, sorry. this should be deferable.

Comment (njohner) [reply opengever/core/upgrades/20190307102922_reindex_dossier_end_date/upgrade.py]: 😓

**Pull request merged**